components/movementsensor/gpsrtk/gpsrtk.go

```diff
@@ -245,28 +246,35 @@ func newRTKMovementSensor(
 	if err := g.Start(ctx); err != nil {
 		return nil, err
 	}
+	g.mu.Lock()
+	defer g.mu.Unlock()
 	return g, g.lastError
 }
 
 func (g *RTKMovementSensor) setLastError(err error) {
-	g.errMu.Lock()
-	defer g.errMu.Unlock()
+	g.mu.Lock()
+	defer g.mu.Unlock()
 
 	g.lastError = err
 }
 
 // Start begins NTRIP receiver with specified protocol and begins reading/updating MovementSensor measurements.
 func (g *RTKMovementSensor) Start(ctx context.Context) error {
```

Comment:
While we're fixing issues here and this may be dumb... do we actually use `RTKMovementSensor`? I only see `newRTKMovementSensor` called in tests. Furthermore, the constructor calls `Start` which has a context that is passed into `newRTKMovementSensor` which is incorrect compared to passing in the `cancelCtx`. Furthermore, the cancel context is created with the context passed into new which is also wrong. It should be a standalone background context.

Reply:
Is your issue the public struct? we can make it private. 

I remember that context was something that got the driver working way back when. However, this package needs a bit of cleaning but is not easily testable in real contexts as it needs proximity to a ntrip mountpoint and a wide open sky. A spring cleaning task for me.

Reply:
My issue is what is this code for. There's no way to use it as a user

Reply:
kind of, this sensor has two physical things communicating. There's a GPS chip that wants to get satellite data and use correction data from a source to get sub 2cm positioning corrections. 

There is a source of these corrections that is also set up in this driver or a separate model, that starts a server or i2c/serial correction from a secondary chip that you can hoist on an antenna by yourself.

It is ugly but it works. Can it be cleaned up? Yes. Do we need to care about cleaning it up right now? No. Was the flaky test due to a mountpoint time out on the ntrip external library and not something we wrote, yes. I'd stick to just fixing that and the locks for this PR.

Reply:
Yeah I'm just confused why we're testing code that isn't in use anywhere. Regarding contexts, for a constructor specifically, a context is used at request time and for any startup related routines that run before construction returns